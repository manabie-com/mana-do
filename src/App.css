.App {
  text-align: center;
  display: flex;
  justify-content: center;
}
