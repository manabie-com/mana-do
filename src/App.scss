@import "./styles/color";

.App {
    text-align: center;
    display: flex;
    justify-content: center;
    height: 100%;
    position: relative;
    padding: 1rem 0.5rem;
}

button {
    outline: none;
    border: none;
    // box-shadow: 2px 0 2px currentColor;
    border-radius: 4px;
    min-height: 32px;
    min-width: 80px;
    padding: 4px 8px;
}

button:hover {
    opacity: 0.85;
}

input[type="checkbox"] {
    width: 24px;
    height: 24px;
    box-shadow: none;
    border: none;
    outline: none;
}

input[type="checkbox"]:focus {
    box-shadow: none;
    border: none;
    outline: none;
}

input {
    min-height: 36px;
    outline: none;
    padding: 0 12px;
    border-radius: 4px;
}

input:focus {
    box-shadow: 0 0 0 0.25rem rgba(13, 110, 253, 0.4);
    transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
    border-color: #86b7fe;
}
