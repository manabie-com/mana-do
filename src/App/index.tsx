import React from "react";
import Routes from "routes";
import GlobalStyle from "global.styles";

function App() {
  return (
    <>
      <GlobalStyle />
      <Routes />
    </>
  );
}

export default App;
