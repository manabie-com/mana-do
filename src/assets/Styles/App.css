.App {
  text-align: center;
  display: flex;
  justify-content: center;
  max-width: 70vw;
  gap:14px;
  margin:20px auto;
  max-height: calc(100vh - 40px);
}

.bar {
  background-color: #E5E3C9;
}

.bar__left{
  display: flex;
  width: 30%;
  flex-direction: column;
  gap: 20px;
  padding: 20px;
  height: calc( 100vh - 40px - 40px );
}
.bar__right{
  padding:20px;
  width: 70%;
  display: flex;
  flex-direction: column;
  gap: 20px;
  position: relative;
}

button {
  outline: none;
  border: none;
  border-radius: 4px;
  min-height: 32px;
  min-width: 80px;
  padding: 4px 8px;
  cursor: pointer;
}

button:hover {
  opacity: 0.85;
}

.btn{
  border-radius: 4px;
  text-shadow: 0 1px 1px rgba(0, 0, 0, 0.2);
  cursor: pointer;
}

.btn-success {
  background: rgb(28, 184, 65);
}

.btn-error {
  background: rgb(202, 60, 60);
}

.btn-warning {
  background: rgb(223, 117, 20);
}

.btn-primary {
  background: rgb(66, 184, 221);
}

input[type="checkbox"] {
  width: 24px;
  height: 24px;
  box-shadow: none;
  /*border: none;*/
  /*outline: none;*/
  border-radius: 100%;
  cursor: pointer;
}

input[type="checkbox"]:focus {
  box-shadow: none;
  border: none;
  outline: none;
}

input {
  min-height: 36px;
  border: none;
  outline: none;
  padding: 0 12px;
  box-shadow: 2px 0 4px rgba(0,0,0, 0.2);
  border-radius: 4px;
}

input:focus {
  box-shadow: 1px 0 9px rgba(0,0,0, 0.25);
}

.ToDo__container {
  background-color: #E5E3C9;
  border: 1px solid rgba(0,0,0, 0.13);
  border-radius: 8px;
  width: 500px;
  margin-top: 5rem;
  padding: 24px;
  box-shadow: 2px 2px 1px rgba(0,0,0, 0.09),
              3px 2px 3px rgba(0,0,0, 0.05);
}

.Todo__creation {
  display: flex;
}

.Todo__input {
  flex: 1 1;
}

.ToDo__list {
  display: flex;
  flex-direction: column;
  margin-top: 1.5rem;
}

.ToDo__item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #ffffff80;
  margin: 20px 8px;
  padding: 0 8px;
  border-radius: 4px;
  cursor: pointer;
}

.ToDo__item > span {
  flex: 1 1;
  text-align: left;
  margin-left: 8px;
}

.Todo__content {
  flex: 1 1;
}

.Todo__delete {
  outline: none;
  border: none;
  width: 32px;
  height: 32px;
  min-width: auto;
  min-height: auto;
  box-shadow: none;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}

.Todo__action, .Todo__delete {
  width: 24px;
  height: 24px;
  flex-shrink: 0;
}

.Todo__toolbar {
  display: flex;
  justify-content: space-between;
  margin-top: 24px;
}

.Todo__tabs {
  display: flex;
  justify-content: center;
}

.Todo__tabs > *:not(:last-child) {
  margin-right: 8px;
}

.Action__btn {
}

.lds-dual-ring {
  display: inline-block;
  width: 80px;
  height: 80px;
  position: absolute;
  top: 50%;
  right: 50%;
}
.lds-dual-ring:after {
  content: " ";
  display: block;
  width: 64px;
  height: 64px;
  margin: 8px;
  border-radius: 50%;
  border: 6px solid #789395;
  border-color: #789395 transparent #789395 transparent;
  animation: lds-dual-ring 1.2s linear infinite;
}
@keyframes lds-dual-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.WrapperItemSiderBar{
  padding: 12px;
  text-align: left;
  border-radius: 6px;
}

.WrapperItemSiderBar:hover{
  opacity: 0.85;
  transition: 0.5s;
  cursor: pointer;
  background-color: #80808021;
}

.ItemSiderBar{
  padding-left: 8px;
  display: flex;
  gap: 16px;
  align-items: center;
  transition: .1s;
}

.ItemSiderBar--acitve{
  border-left: 4px solid #789395;
  transition: .1s;
}

.todo__list{
  height: 95%;
  overflow: auto;
}

.todo__add__task{
}

.ToDo__item:hover{
  background-color: #ffffff;
}
.Wrapper__GroupActionTodo{
  display: flex;
  gap: 10px;
}

.AddTask {
  display: flex;
  gap:20px
}