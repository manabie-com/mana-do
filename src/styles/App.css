.App {
  text-align: center;
  display: flex;
  justify-content: center;
}

button {
  outline: none;
  border: none;
  box-shadow: 2px 0 2px currentColor;
  border-radius: 4px;
  min-height: 32px;
  min-width: 80px;
  padding: 4px 8px;
}

button:hover {
  opacity: 0.85;
}

input[type="checkbox"] {
  width: 24px;
  height: 24px;
  box-shadow: none;
  border: none;
  outline: none;
}

input[type="checkbox"]:focus {
  box-shadow: none;
  border: none;
  outline: none;
}

input {
  min-height: 36px;
  border: none;
  outline: none;
  padding: 0 12px;
  box-shadow: 2px 0 4px rgba(0,0,0, 0.2);
  border-radius: 4px;
}

input:focus {
  box-shadow: 1px 0 9px rgba(0,0,0, 0.25);
}

.ToDo__container {
  border: 1px solid rgba(0,0,0, 0.13);
  border-radius: 8px;
  width: 500px;
  margin-top: 5rem;
  padding: 24px;
  box-shadow: 2px 2px 1px rgba(0,0,0, 0.09),
              3px 2px 3px rgba(0,0,0, 0.05);
}

.Todo__creation {
  display: flex;
}

.Todo__input {
  flex: 1 1;
}

.ToDo__list {
  display: flex;
  flex-direction: column;
  margin-top: 1.5rem;
}

.ToDo__item {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.ToDo__item > span {
  flex: 1 1;
  text-align: left;
  margin-left: 8px;
}

.Todo__content {
  flex: 1 1;
}

.Todo__delete {
  outline: none;
  border: none;
  width: 32px;
  height: 32px;
  min-width: auto;
  min-height: auto;
  box-shadow: none;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}

.Todo__action, .Todo__delete {
  width: 24px;
  height: 24px;
  flex-shrink: 0;
}

.Todo__toolbar {
  display: flex;
  justify-content: space-between;
  margin-top: 24px;
}

.Todo__tabs {
  display: flex;
  justify-content: center;
}

.Todo__tabs > *:not(:last-child) {
  margin-right: 8px;
}

.Action__btn {
}