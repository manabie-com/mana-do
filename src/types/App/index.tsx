export interface AppProps {
    token: string
}