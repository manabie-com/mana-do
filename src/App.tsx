import React from 'react';

import Modules from './Modules'

import './assets/Styles/App.css';
import './assets/Icons/style.css'

function App() {
  return (
    <main className="App" data-testid="App">
        <Modules></Modules>
    </main>
  );
}

export default App;
