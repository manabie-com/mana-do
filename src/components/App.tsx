import React from 'react';

import ToDoPage from './ToDoPage';

import '../styles/App.css';

function App() {
  return (
    <main className="App">
      <ToDoPage />
    </main>
  );
}

export default App;
