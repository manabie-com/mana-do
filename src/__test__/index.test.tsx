import React from 'react';
import App from "../App";
import {render} from "@testing-library/react";

describe('<App /> render', () => {
    it('should match snapshot', function () {
        const elm = render(<App/>);
        expect(elm.container.firstChild).toMatchSnapshot();
    });
})