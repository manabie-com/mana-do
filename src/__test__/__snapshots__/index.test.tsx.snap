// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render should match snapshot 1`] = `
<main
  class="App"
  data-testid="App"
>
  <div
    class="bar bar__left"
  >
    <div
      class="WrapperItemSiderBar"
    >
      <div
        class="ItemSiderBar--acitve ItemSiderBar"
      >
        <i
          class="manabie-endless"
        />
        <span>
          All
        </span>
      </div>
    </div>
    <div
      class="WrapperItemSiderBar"
    >
      <div
        class="false ItemSiderBar"
      >
        <i
          class="manabie-active"
        />
        <span>
          Active
        </span>
      </div>
    </div>
    <div
      class="WrapperItemSiderBar"
    >
      <div
        class="false ItemSiderBar"
      >
        <i
          class="manabie-phong"
        />
        <span>
          Completed
        </span>
      </div>
    </div>
    <div
      class="WrapperItemSiderBar"
    >
      <div
        class="false ItemSiderBar"
      >
        <i
          class="manabie-star_outline"
        />
        <span>
          Important
        </span>
      </div>
    </div>
    <div
      class="WrapperItemSiderBar"
    >
      <div
        class="false ItemSiderBar"
      >
        <i
          class="manabie-remove"
        />
        <span>
          Clear all todos
        </span>
      </div>
    </div>
  </div>
  <div
    class="bar bar__right"
  >
    <div
      class="lds-dual-ring"
    />
    <div
      class="todo__list"
    />
    <div
      class="AddTask"
    >
      <input
        aria-label="todo-input"
        class="Todo__input"
        data-testid="todo-input"
        placeholder="+ Add a task"
      />
    </div>
  </div>
</main>
`;
